src/main/java/org/sonar/plugins/jacoco/ReportPathsProvider.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarQube JaCoCo Plugin
+ * Copyright (C) 2018-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.jacoco;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.sensor.SensorContext;
+
+class ReportPathsProvider {
+  private static final String[] DEFAULT_PATHS = {"target/site/jacoco/jacoco.xml", "build/reports/jacoco/test/jacocoTestReport.xml"};
+  private static final String REPORT_PATHS_PROPERTY_KEY = "sonar.coverage.jacoco.xmlReportPaths";
+  private final SensorContext context;
+
+  ReportPathsProvider(SensorContext context) {
+    this.context = context;
+  }
+
+  Collection<Path> getPaths() {
+    Set<Path> reportPaths = new HashSet<>(Arrays.asList(context.config().getStringArray(REPORT_PATHS_PROPERTY_KEY)))
```

Comment:
`Arrays.asList(...).stream()` can be replaced with `Stream.of` or `Arrays.stream`

Reply:
Is it on purpose that we don't check if a report exists? (like below for `DEFAULT_PATHS`)
A test case that verifies the intention would be good too. (For example, what should happen if paths were specified but none of them exist. Should we fall back to default paths, or use empty list?)

Reply:
The error will happen later when trying to read the report, and fail the analysis. The point is that if you are specifying the path then the users is expecting for it to be there and we should fail fast if it's not there.